plugins/pipewire/PipeWireDevice.h

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright 2009-2024 Jörg Müller
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+#pragma once
+
+#ifdef PIPEWIRE_PLUGIN
+#define AUD_BUILD_PLUGIN
+#endif
+
+/**
+ * @file PipeWireDevice.h
+ * @ingroup plugin
+ * The PipeWireDevice class.
+ */
+
+#include <pipewire/pipewire.h>
+
+#include "devices/SoftwareDevice.h"
+
+AUD_NAMESPACE_BEGIN
+
+/**
+ * This device plays back through PipeWire, the simple direct media layer.
+ */
+class AUD_PLUGIN_API PipeWireDevice : public SoftwareDevice
+{
+private:
+	class PipeWireSynchronizer : public DefaultSynchronizer
+	{
+		PipeWireDevice* m_device;
+		int64_t m_tick_start = 0.0f;
+		double m_seek_pos = 0.0f;
+		double m_timeline_pos = 0.0f;
+
+	public:
+		PipeWireSynchronizer(PipeWireDevice* device);
+
+		void update_tick_start();
+		virtual void seek(std::shared_ptr<IHandle> handle, double time);
+		virtual double getPosition(std::shared_ptr<IHandle> handle);
+	};
+
+	/// Synchronizer.
+	PipeWireSynchronizer m_synchronizer;
+
+	/**
+	 * Whether there is currently playback.
+	 */
+	bool m_playback;
+
+	pw_stream* m_stream;
+	pw_thread_loop* m_thread;
+	std::unique_ptr<pw_stream_events> m_events;
+
+	AUD_LOCAL static void handle_state_changed(void* device_ptr, enum pw_stream_state old, enum pw_stream_state state, const char* error);
+	/**
+	 * Mixes the next bytes into the buffer.
+	 * \param data The PipeWire device.
+	 */
+	AUD_LOCAL static void mix_audio_buffer(void* device_ptr);
```

Comment:
Please stick with audaspace's camelCase naming convention for methods when possible.